lhotse/cut/set.py

```diff
@@ -2550,6 +2551,306 @@ def prefetch(self, buffer_size: int = 10) -> "CutSet":
             )
         )
 
+    """
+    ╔══════════════════════════════════════╗
+    ║ Export CutSet to HuggingFace Dataset ║
+    ╚══════════════════════════════════════╝
+    """
+
+    def contains_only_mono_cuts(self) -> bool:
+        return all(isinstance(cut, MonoCut) for cut in self)
+
+    def has_one_supervision_per_cut(self) -> bool:
+        return all(len(cut.supervisions) == 1 for cut in self)
+
+    def has_one_audio_source(self) -> bool:
+        return all(len(cut.recording.sources) == 1 for cut in self)
+
+    def _convert_cuts_info_to_hf(self) -> Tuple[Dict[str, Any], Dict[str, Any]]:
```

Comment:
Since this file is pretty large and the functionality introduced is fairly well isolated, I suggest to change every method into a function (except the main HF export API) and move them into a separate file `lhotse/hf.py`. Keep the HF `datasets` imports local like they are. The body of `to_huggingface_dataset(self)` should import and call `def export_cuts_to_hf(cuts)` from that file.

Reply:
Done.